src/serac/physics/state/state_manager.hpp

```diff
@@ -95,19 +95,119 @@ class StateManager {
+          geom_group->createViewScalar("total_size", total_size);
+
+          // Tell Sidre where the external array is, how large it is (calculated above), and what is in it (faking
+          // uint8)
+          axom::sidre::View* states_view = geom_group->createView("states");
+          states_view->setExternalDataPtr(axom::sidre::UINT8_ID, num_uint8s, states.data());
+        } else {
+          // Get Sidre group of where the states were stored.
+          // Note: this data is not owned by Sidre and the array should have been created at this point but
+          // the previous data has not been loaded yet into the array.
+          SLIC_ERROR_ROOT_IF(!bp_group->hasGroup(qds_group_name),
+                             axom::fmt::format("Loaded Sidre Datastore did not have group for Quadrature Datas"));
+          axom::sidre::Group* qdatas_group = bp_group->getGroup(qds_group_name);
+          SLIC_ERROR_ROOT_IF(
+              !qdatas_group->hasGroup(std::string(geom_name)),
+              axom::fmt::format("Loaded Sidre Datastore did not have group for Quadrature Data geometry type '{}'",
+                                std::string(geom_name)));
+          axom::sidre::Group* geom_group = qdatas_group->getGroup(std::string(geom_name));
+
+          // Verify size correctness
+          auto verify_size = [](axom::sidre::Group* group, int value, const std::string& view_name,
+                                const std::string& err_msg) {
+            SLIC_ERROR_IF(
+                !group->hasView(view_name),
+                axom::fmt::format("Loaded Sidre Datastore does not have value '{}' for Quadrature Data.", view_name));
+            auto prev_value = group->getView(view_name)->getData<axom::IndexType>();
+            SLIC_ERROR_IF(value != prev_value, axom::fmt::format(err_msg, value, prev_value));
+          };
+          verify_size(geom_group, num_states, "num_states",
+                      "Current number of Quadrature Data States '{}' does not match value in restart '{}'.");
+          verify_size(geom_group, state_size, "state_size",
+                      "Current size of Quadrature Data State '{}' does not match value in restart '{}'.");
+          verify_size(geom_group, total_size, "total_size",
+                      "Current total size of Quadrature Data States '{}' does not match value in restart '{}'.");
+
+          // Tell Sidre where the external array is
+          SLIC_ERROR_ROOT_IF(!geom_group->hasView("states"),
+                             "Loaded Quadrature Data geometry Sidre view did not have 'states'");
+          axom::sidre::View* states_view = geom_group->getView("states");
+          states_view->setExternalDataPtr(states.data());
+        }
+      }
+    }
+
+    if (is_restart_) {
+      // NOTE: This call will reload all external buffers from file stored in the DataStore
+      // TODO: This should be changed to load only the current material quadrature data after
```

Comment:
might be good to point to an issue either in serac or another proj to track this

Reply:
oh man.. thanks for reminding me of this. This is no longer true